front/components/assistant/RenderMessageMarkdown.tsx

```diff
@@ -331,7 +338,18 @@ function CiteBlock(props: ReactMarkdownProps) {
       <span className="inline-flex space-x-1">
         {refs.map((r, i) => {
           const document = references[r.ref];
-          const link = makeLinkForRetrievedDocument(document);
+          let link: string | null = null;
+
+          if (isRetrievalDocumentType(document)) {
+            link = makeLinkForRetrievedDocument(document);
+          } else if (isWebsearchResultType(document)) {
```

Comment:
I guess if you go with a else block here, it should work as well. and you can remove the unreachable code below.